packages/react-form-state/src/FormState.ts

```diff
@@ -424,11 +458,17 @@ function createFormState<Fields>(values: Fields): State<Fields> {
     };
   });
 
+  let fieldsWithErrors = fields;
+  if (externalErrors) {
```

Comment:
A couple things:
- the let, if, reassignment pattern isn't really something this library does anywhere else so this feels a little weird in terms of coding style
- we need to also handle filtering down errors into the fields in `reconcileFormState`
- we could probably simplify this stuff if we only push the errors down into the fields at `formData()`, then `getDerivedStateFromProps` doesn't care about `externalErrors` at all and we don't need to store this stuff in state.

Reply:
- Do you suggest two return statement instead?
- Can you elaborate on `we need to also handle filtering down errors into the fields in reconcileFormState`?
- Agree. Maybe my understanding of FormState is unclear.
See https://github.com/Shopify/quilt/pull/461/files#r246853227

Reply:
* Two return statements or a ternary assignment is more in line with the rest of the code (and most of our react code in general)

* If is set to `reset-where-changed` when initial values change it's state derivation logic will be using`reconcileFormState`. As such we need to facilitate that by having the same error pass-down logic there too

*Spoke to @darrenhebner about this on slack, we definitely need this in state due to it being independently clearable on blur of a field. Responded to your other comment as well, but we definitely are a component that needs derived state (one could argue we should maybe be fully uncontrolled instead, but that would be a massive breaking api change, and probably an opt-in mode) and `componentDidUpdate` has no benefits for us over `getDerivedStateFromProps`.